xdsl/backend/csl/print_csl.py

```diff
@@ -29,23 +29,27 @@ class CslPrintContext:
 
     _prefix: str = field(default="")
 
-    def print(self, text: str, prefix: str = ""):
+    def print(self, text: str, prefix: str = "", end: str = "\n"):
         """
         Print `text` line by line, prefixed by self._prefix and prefix.
         """
         for l in text.split("\n"):
-            print(self._prefix + prefix + l, file=self.output)
+            print(self._prefix + prefix + l, file=self.output, end=end)
 
-    def _get_variable_name_for(self, val: SSAValue) -> str:
+    def _get_variable_name_for(self, val: SSAValue, hint: str | None = None) -> str:
```

Comment:
I'm not sure how much we want to rely on name hints for naming variables in the output, especially now that we mess with them at parsing time, like stripping `_0`, `_1`, ... suffixes. The two alternatives that come to mind is either a custom type for variables or an attribute on ops that own the variables. The second seems vaguely preferable. I don't want to derail your csl mergeathon too much, but if you agree that not relying on name hints is a good change in the future, it would be great to file a PR and mention it in a comment.

Reply:
Yeah, not sure how we want to continue to go on with name hints from here. I'd see how it evolves over time, maybe we don't actually need readable CSL (the official examples don't seem to bother too much with that either).